api/store/mongo/device_store.go

```diff
@@ -450,8 +450,8 @@ func (s *Store) DeviceDeleteTag(ctx context.Context, uid models.UID, tag string)
 	return err
 }
 
-func (s *Store) DeviceRenameTag(ctx context.Context, uid models.UID, currentTagName string, newTagName string) error {
-	_, err := s.db.Collection("devices").UpdateOne(ctx, bson.M{"uid": uid, "tags": currentTagName}, bson.M{"$set": bson.M{"tags.$": newTagName}})
+func (s *Store) DeviceRenameTag(ctx context.Context, currentTagName string, newTagName string) error {
+	_, err := s.db.Collection("devices").UpdateMany(ctx, bson.M{"tags": currentTagName}, bson.M{"$set": bson.M{"tags.$": newTagName}})
```

Comment:
We need to guarantee it needs to be atomic. This needs to be done using a transaction.